cirq-pasqal/cirq_pasqal/pasqal_sampler.py

```diff
@@ -114,8 +133,11 @@ def run_sweep(
             Result list for this run; one for each possible parameter
             resolver.
         """
-        assert isinstance(program.device, cirq_pasqal.PasqalDevice)
-        program.device.validate_circuit(program)
+        if program._device != cirq.UNCONSTRAINED_DEVICE:
+            assert isinstance(program._device, cirq_pasqal.PasqalDevice)
```

Comment:
I would either assert with a message or raise ValueError.  Since this is a user specified parameter and not an invariant, I think the latter would be more appropriate.

Reply:
I'm keeping this here from the original code, would we prefer to make the break here ?

Reply:
Let's add a message then to keep with existing assertion behavior.

Reply:
Done.